### ISSUE TITLE
Runtime Error on querying refseq_select_prot

### ISSUE THREAD

**Issue by emilhaegglund** [bbuchfink/diamond#505]
Hi,
Very nice with the possibility to use precompiled BLAST databases. However, when using the refseq_select_prot database downloaded from NCBI FTP I get a Runtime Error. 

I used the following commands to prepare the db, which worked fine for the refseq_protein database.
```
update_blastdb.pl --decompress --blastdb_version 5 refseq_select_prot
./diamond prepdb -d ../data/refseq_select_prot/refseq_select_prot
```
And this is the error message
```diamond v2.0.11.149 (C) Max Planck Society for the Advancement of Science
Documentation, support and updates available at http://www.diamondsearch.org
Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)

#CPU threads: 32
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory:
#Target sequences to report alignments for: 25
Opening the database...  [0.003s]
Database: ../data/refseq_select_prot/refseq_select_prot (type: BLAST database, sequences: 53363808, letters: 17493288360)
Block size = 2000000000
Opening the input file...  [0s]
Opening the output file...  [0s]
Loading query sequences...  [0s]
Masking queries...  [0.004s]
Building query seed set...  [0s]
Algorithm: Query-indexed
Building query histograms...  [0.001s]
Allocating buffers...  [0s]
Loading reference sequences...  [3.13s]
Initializing dictionary...  [0.011s]
Initializing temporary storage...  [0s]
Building reference histograms... terminate called after throwing an instance of 'std::runtime_error'
  what():  Unrecognized sequence character in BLAST database (0, id=989838, pos=295)
Aborted (core dumped)
```

I also used the blastdbcmd to check the sequence with id 989838 but it can't find that entry in the refseq_select_prot database.

Cheers, Emil

**Comment by bbuchfink** [comment]
The error seems to be caused by sequences that have gap letters in them e.g.:
```
>WP_034549049.1 anthranilate phosphoribosyltransferase [Helicobacter fennelliae]
MAIKKAIIELTQKHNLSQDMAFAVMNEIMQGLASPVQISAYLTALSMKGENIDEITASALSMRAHCVRLLHDMDALEIVG
TGGDHSNSFNISTTAALIVSSAGVKVAKHGNRAASSKCGAADVLEALGVNITIPPEKSKELLESINICFLFAQNYHIAMK
YVAPIRKELGIRTIFNILGPLSNPAGANMELLGVYDESLVEPMAKVMQNLGVKRGMVVYGMDKIDEISLSSETKIAEINK
NELKFYTFHPRDAGFSLCDKSDLVGGDATQNAKIIREILEGKEKGAKRDVATLNAGAALYIAGMAKSIESGVRMAEKQID
NKLALAQLERFIKESNK-
```

I can commit a hotfix for this into the master branch, but you'd have to compile from source to use it, if that's ok.

**Comment by emilhaegglund** [comment]
That would be perfect! Thanks in advance.

Cheers,
Emil

**Comment by bbuchfink** [comment]
This should do the trick: f835225a4db0f6adbde2fe33626a35fd20d852e1

**Comment by bbuchfink** [comment]
The fix has now been included in v2.0.12.
